java/src/jmri/jmrit/display/IndicatorTrackPaths.java

```diff
@@ -115,7 +117,9 @@ public void removeLocoIcon() {
         }
     }
 
-    protected void setLocoIcon(OBlock block, Point pt, Dimension size, Editor ed) {
+    @jmri.InvokeOnLayoutThread
```

Comment:
It would be better to use `@jmri.InvokeOnGuiThread`. The layout thread is currently the same as the GUI thread, but in a distant future, they could be separated.